src/lib/writer.ts

```diff
@@ -196,29 +196,63 @@ class Writer {
    *
    * @param buffer Contents of the octet string.
    */
-  writeVarOctetString (buffer: Buffer): void {
-    if (!Buffer.isBuffer(buffer)) {
+  writeVarOctetString (buffer: Buffer | Writer): void {
```

Comment:
Does the Predictor's `writeVarOctetString` need to take a `Writer` also, so that the API's stay matched up?